File: StructuralPatterns/Bridge/TypeScript/Square.ts
import { Shape } from './Shape';
import { Renderer } from './Renderer';

export class Square extends Shape {
    private readonly size: number;
    constructor(renderer: Renderer, size: number) {
        super(renderer);
        this.size = size;
    }
    draw(): void {
        this.renderer.renderSquare(this.size);
    }
}

File: StructuralPatterns/Bridge/Go/square.go
package main

type Square struct {
	renderer Renderer
	size     float64
}

func NewSquare(renderer Renderer, size float64) *Square {
	return &Square{renderer: renderer, size: size}
}

func (s *Square) Draw() {
	s.renderer.RenderSquare(s.size)
}
